tests/lib/rules/no-extra-parens.js

```diff
@@ -545,8 +566,6 @@ ruleTester.run("no-extra-parens", rule, {
         invalid("bar((function(){}).foo(), 0);", "bar(function(){}.foo(), 0);", "FunctionExpression"),
         invalid("bar[(function(){}).foo()];", "bar[function(){}.foo()];", "FunctionExpression"),
         invalid("var bar = (function(){}).foo();", "var bar = function(){}.foo();", "FunctionExpression"),
-        invalid("((function(){}).foo.bar)();", "(function(){}.foo.bar)();", "FunctionExpression", null, { options: ["functions"] }),
-        invalid("((function(){}).foo)();", "(function(){}.foo)();", "FunctionExpression", null, { options: ["functions"] }),
```

Comment:
These two cases were moved into "valid" because the `functions` option is enabled and the parens around the function are necessary (it's the parens around the call expression that are unnecessary). This is consistent with cases like [this](https://github.com/eslint/eslint/blob/a9a81024c166e6eaeb944b37d6b8eb85170c9316/tests/lib/rules/no-extra-parens.js#L558), where the `MemberExpression` parens are reported and removed instead of the `FunctionExpression` parens.